protocol/messenger_communities.go

```diff
@@ -1869,3 +1869,149 @@ func (m *Messenger) RequestExtractDiscordChannelsAndCategories(filesToImport []s
+		}
+	}
+
+	messagesToSave := make([]*common.Message, 0)
+
+	for _, exportedChannel := range exportedData {
+
+		communityChat := &protobuf.CommunityChat{
+			Permissions: &protobuf.CommunityPermissions{
+				Access: protobuf.CommunityPermissions_NO_MEMBERSHIP,
+			},
+			Identity: &protobuf.ChatIdentity{
+				DisplayName: exportedChannel.Channel.Name,
+				Emoji:       "😎",
+				Description: exportedChannel.Channel.Description,
+			},
+			CategoryId: addedCategoriesIds[exportedChannel.Channel.CategoryID],
+		}
+
+		r, err := m.CreateCommunityChat(discordCommunity.ID(), communityChat)
+		if err != nil {
+			return nil, err
+		}
+		response.AddCommunities(r.Communities())
+
+		channelId := ""
+
+		// This looks like we keep overriding the `channelId` but in practice,
+		// this iterates only once
+		addedChatChange := r.CommunityChanges[0]
+		for chatId, _ := range addedChatChange.ChatsAdded {
+			channelId = chatId
+		}
+
+		for _, discordMessage := range exportedChannel.Messages {
+			exists, err := m.persistence.HasDiscordMessageAuthor(discordMessage.Author.GetId())
+			if err != nil {
+				return nil, err
+			}
+			if !exists {
+				err := m.persistence.SaveDiscordMessageAuthor(discordMessage.Author)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			payload, err := proto.Marshal(discordMessage)
+			if err != nil {
+				return nil, err
+			}
+
+			// Creating a `common.RawMessage` to get a `messageID`
+			rawMessage := &common.RawMessage{
+				Payload:        payload,
+				MessageType:    protobuf.ApplicationMetadataMessage_CHAT_MESSAGE,
+				Sender:         m.identity,
+				SkipEncryption: true,
+			}
+
+			messageID, err := m.sender.GetMessageID(rawMessage)
+			if err != nil {
+				return nil, err
+			}
+
+			layout := "2006-01-02T15:04:05+00:00"
+			timestamp, err := time.Parse(layout, discordMessage.Timestamp)
+			if err != nil {
+				return nil, err
+			}
+			discordMessage.Timestamp = fmt.Sprintf("%d", timestamp.Unix())
+
+			if discordMessage.TimestampEdited != "" {
+				timestampEdited, err := time.Parse(layout, discordMessage.TimestampEdited)
+				if err != nil {
+					return nil, err
+				}
+				discordMessage.TimestampEdited = fmt.Sprintf("%d", timestampEdited.Unix())
+			}
+
+			messageToSave := &common.Message{
+				ID:               types.EncodeHex(messageID),
+				WhisperTimestamp: uint64(timestamp.Unix()),
+				From:             types.EncodeHex(crypto.FromECDSAPub(&m.identity.PublicKey)),
+				Seen:             true,
+				LocalChatID:      channelId,
+				SigPubKey:        &m.identity.PublicKey,
+				CommunityID:      discordCommunity.IDString(),
+				ChatMessage: protobuf.ChatMessage{
+					MessageType: protobuf.MessageType_COMMUNITY_CHAT,
+					ContentType: protobuf.ChatMessage_DISCORD_MESSAGE,
+					Clock:       m.getTimesource().GetCurrentTime(),
+					ChatId:      channelId,
+					Payload: &protobuf.ChatMessage_DiscordMessage{
+						DiscordMessage: discordMessage,
+					},
+				},
+			}
```

Comment:
Here, I was wondering if there are any fields in `common.Message` that are super mandatory but not set here.